File: 10-07/3/ac/sol.py
def main():
    m, n = map(int, input().split())
    assert(1 <= m <= 1000)
    assert(1 <= n <= 1000)
    ia = list(map(int, input().split()))
    assert(len(ia) == m + 1)
    a = list(input().split())
    assert(len(a) == ia[-1])
    ja = list(map(int, input().split()))
    assert(len(ja) == ia[-1])

    a_ptr = 0
    for row_ind in range(m):
        row = ['0'] * n
        num_in_row = ia[row_ind + 1] - ia[row_ind]
        for _ in range(num_in_row):
            row[ja[a_ptr]] = a[a_ptr]
            a_ptr += 1

        print(' '.join(row))

t = int(input())
for _ in range(t):
    main()

File: 10-07/3/ac/sol.cpp
#include <iostream>
#include <iomanip>
#include <cstring>
#include <algorithm>
#include <utility>

using namespace std;

int m, n;
int matrix[1003];
int a[1003 * 1003];
int ia[1003];
int ja[1003 * 1003];

void read_input() {
    cin >> m >> n;
    for (int i = 0; i <= m; ++i) {
        cin >> ia[i];
    }
    int l = ia[m];
    for (int i = 0; i < l; ++i) {
        cin >> a[i];
    }
    for (int i = 0; i < l; ++i) {
        cin >> ja[i];
    }
}

int main() {
    int T;
    cin >> T;
    while (T-- > 0) {
        read_input();

        int a_ptr = 0;

        for (int row = 0; row < m; ++row) {
            memset(matrix, 0, sizeof(matrix));
            int num_in_row = ia[row + 1] - ia[row];

            for (int k = 0; k < num_in_row; ++k) {
                matrix[ja[a_ptr]] = a[a_ptr];
                a_ptr++;
            }

            cout << matrix[0];
            for (int col = 1; col < n; ++col) {
                cout << ' ' << matrix[col];
            }
            cout << '\n';
        }
    }

    return 0;
}
